include/revng/Pipes/CompileModule.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/ContainerSet.h"
+#include "revng/Pipeline/InputOutputContract.h"
+#include "revng/Pipeline/InspectableLLVMContainer.h"
+#include "revng/Pipeline/LLVMContainer.h"
+#include "revng/Pipeline/LLVMPipe.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/IsolatedKind.h"
+#include "revng/Pipes/RevngPipes.h"
+#include "revng/Pipes/RootKind.h"
+#include "revng/Pipes/TemporaryFile.h"
+
+namespace Pipeline {
+
+class CompileModulePipe {
+public:
+  static constexpr auto Name = "Compile Module Pipe";
+  std::array<AtomicContract, 1> getContract() const {
+    return { AtomicContract(Root, Exactness::DerivedFrom, 0, Object, 1, true) };
```

Comment:
What do you think about having a set of support `Contract` classes that cover the most common use case with intuitive names?
```suggestion
    return { UpgradeTargetContract(Root, Object) };
```

Reply:
i wanted to do this, i just never had the time to stop and think which are the ones that one uses 90% of the time

Reply:
We have some real-world users, so it might be a good moment to do that.

Reply:
i will delay this for a future merge request otherwise we get lost in the details